soh/src/code/z_parameter.c

```diff
@@ -2434,6 +2454,9 @@ u16 Randomizer_Item_Give(GlobalContext* globalCtx, GetItemEntry giEntry) {
                 gSaveContext.inventory.dungeonKeys[mapIndex]++;
                 return RG_NONE;
             }
+        } else if ((item >= RG_FOREST_TEMPLE_KEY_RING) && (item <= RG_GANONS_CASTLE_KEY_RING)) {
+            gSaveContext.inventory.dungeonKeys[mapIndex] = maxKeys;
```

Comment:
looks like `maxKeys` is only being used here if we're dealing with keyrings, i guess the logic to set that value could be moved here but adding another switch/case feels messy.

maybe i'm just not a fan of `maxKeys` as a name, might be nice to use something more verbose that implies it is only used to set the keys in the inventory when getting a keyring

only names that come to mind are `keysOnKeyring` or `numOfKeysOnKeyring`, but i'm very open to suggestions

Reply:
Yep, reasoning was exactly what you mentioned, I could handle this separately but would have to basically repeat this entire switch statement.

Renamed to `numOfKeysOnKeyring` 👍